templates/lb-fargate-service/cf.yml

```diff
@@ -118,8 +124,74 @@ Resources:
             Action: 'sts:AssumeRole'
-      ManagedPolicyArns:
-        - 'arn:aws:iam::aws:policy/PowerUserAccess'
+      Policies:
+        - PolicyName: 'DenyIAMExceptTaggedRoles'
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: 'Deny'
+                Action: 'iam:*'
+                Resource: '*'
+              - Effect: 'Allow'
+                Action: 'sts:AssumeRole'
+                Resource:
+                  - !Sub 'arn:aws:iam::${AWS::AccountId}:role/*'
+                Condition:
+                  StringEquals:
+                    'iam:ResourceTag/ecs-project': !Sub '${ProjectName}'
+                    'iam:ResourceTag/ecs-environment': !Sub '${EnvName}'
+        - PolicyName: 'AllowPrefixedResources'
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: 'Allow'
+                Action: '*'
+                Resource:
+                  - !Sub 'arn:aws:s3:::${ProjectName}-${EnvName}-*'
+                  - !Sub 'arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:*/${ProjectName}-${EnvName}-*'
+                  - !Sub 'arn:aws:elasticache:${AWS::Region}:${AWS::AccountId}:*/${ProjectName}-${EnvName}-*'
+                  - !Sub 'arn:aws:redshift:${AWS::Region}:${AWS::AccountId}:*:${ProjectName}-${EnvName}-*'
+                  - !Sub 'arn:aws:rds:${AWS::Region}:${AWS::AccountId}:*:${ProjectName}-${EnvName}-*'
+                  - !Sub 'arn:aws:es:${AWS::Region}:${AWS::AccountId}:*/${ProjectName}-${EnvName}-*'
+
+                  - !Sub 'arn:aws:sns:${AWS::Region}:${AWS::AccountId}:${ProjectName}-${EnvName}-*'
+                  - !Sub 'arn:aws:sqs:${AWS::Region}:${AWS::AccountId}:${ProjectName}-${EnvName}-*'
+                  - !Sub 'arn:aws:kinesis:${AWS::Region}:${AWS::AccountId}:*/${ProjectName}-${EnvName}-*'
+                  - !Sub 'arn:aws:firehose:${AWS::Region}:${AWS::AccountId}:*/${ProjectName}-${EnvName}-*'
+                  - !Sub 'arn:aws:kinesisanalytics:${AWS::Region}:${AWS::AccountId}:*/${ProjectName}-${EnvName}-*'
+        - PolicyName: 'AllowTaggedResources' # See https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_actions-resources-contextkeys.html
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: 'Allow'
+                Action: '*'
+                Resource: '*'
+                Condition:
+                  StringEquals:
+                    'aws:ResourceTag/ecs-project': !Sub '${ProjectName}'
+                    'aws:ResourceTag/ecs-environment': !Sub '${EnvName}'
+              - Effect: 'Allow'
+                Action: '*'
+                Resource: '*'
+                Condition:
+                  StringEquals:
+                    'secretsmanager:ResourceTag/ecs-project': !Sub '${ProjectName}'
+                    'secretsmanager:ResourceTag/ecs-environment': !Sub '${EnvName}'
```

Comment:
I'm assuming the Statement above this one is not covering `secretsmanager`? The only difference seems to be `aws` versus `secretsmanager`?

Reply:
yeah correct, some services don't have the "aws" namespace. I'm not sure why :P I guess it's some legacy reason.

For example, [ecs](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonelasticcontainerservice.html#amazonelasticcontainerservice-resources-for-iam-policies) has both the aws and service namespace.  
On the other hand, [other services](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_awssecretsmanager.html#awssecretsmanager-resources-for-iam-policies) don't have the "aws" namespace.

Reply:
Is it possible to combine the 2?

Reply:
I thought that would create an AND relation and result in matching nothing.

I'll test it out and send a follow-up if it's combinable